geoh5py/ui_json/input_file.py

```diff
@@ -367,6 +385,37 @@ def write_ui_json(
 
         return self.path_name
 
+    def set_data_value(self, data: dict):
```

Comment:
The only place we use this is in the setter_validator on params where it handles a single value at a time.  Maybe just pass key, value in from the caller and ditch the loop?

Reply:
Good point. Done